functionsChallenges.js

```diff
@@ -0,0 +1,74 @@
+function maxOrMin(max, x, y) {
+    if (max && x > y) {
+        return x
+    } else if (max && y > x) {
+        return y
+    }
+}
+alert(maxOrMin(true,9,7))
+
+
+
+
+
+
+var arr = [1,4,6,8,9,7]
+function sillySum (array) {
+	var newArr =[]; 
+	for (var i = 0; i < array.length; i++) {
+		
+		 newArr.push(i*array[i]); 
+
+	}
+	return newArr }
+
+sillySum(arr)
+[0, 4, 12, 24, 36, 35]
+
+
+
+
+
+
+
+var arr = [1,4,6,8,9,7]
+function numSquare (array) {
```

Comment:
We want our function to take in a max number and return an array of squares up to that number. Example:

```js
numSquare(36);
//=> [ 1, 4, 9, 16, 25, 36 ]

numSquare(100);
//=> [ 1, 4, 9, 16, 25, 36, 49, 64, 81, 100 ]
```

Try these steps:

```js
function numSquare(max) {
  // define a variable called `squares` that's initially an empty array
  // write a for loop that starts at `var i = 1` and goes until `i * i <= max`
  // inside the loop, push the square of `i` into the `squares` array
  // after the loop is done, return your squares array
}
```